CMake/CommonSetup.cmake

```diff
@@ -13,7 +13,11 @@ set ( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin )
 set_property ( GLOBAL PROPERTY USE_FOLDERS ON )
 
 if ( CMAKE_COMPILER_IS_GNUCXX )
```

Comment:
Thanks for do this.
Can we swap these `if`s out for the more modern way of doing this like `set(CMAKE_CXX_STANDARD 14)` in [`CommonSetup.cmake`](https://github.com/mantidproject/mantid/blob/3b516d98b0ee6bd06e98ca98cb8b0a5efa08b07e/buildconfig/CMake/CommonSetup.cmake#L241)